driver/compenv.ml

```diff
@@ -629,21 +629,32 @@ type deferred_action =
 let c_object_of_filename name =
   Filename.chop_suffix (Filename.basename name) ".c" ^ Config.ext_obj
 
-let process_action
-    (ppf, implementation, interface, ocaml_mod_ext, ocaml_lib_ext) action =
+type action_context = {
+  log : Format.formatter;
+  compile_implementation:
+    start_from:Clflags.Compiler_pass.t ->
+    source_file:string -> output_prefix:string -> unit;
+  compile_interface:
+    source_file:string -> output_prefix:string -> unit;
+  ocaml_mod_ext: string;
+  ocaml_lib_ext: string;
+}
+
+let process_action ctx action =
```

Comment:
Pattern matching on `ctx` could be used here since `ctx` itself is not reused anywhere and all fields are used.

Reply:
Done, thanks!